app/models/StatusTransition.scala

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+import models.DepartureStatus._
+
+object StatusTransition {
+
+  val validTransitions = Set(
+    // Initialized transitions
+    Initialized -> DepartureSubmitted,
+    Initialized -> MrnAllocated,
+    Initialized -> DepartureRejected,
+    Initialized -> PositiveAcknowledgement,
```

Comment:
@crystalmark after speaking with @RobertLaverick he suggested making it valid to transition from `Initialized` to `PositiveAcknowledgement` status for self-healing purposes but we thought it would be worth running that past you

Reply:
I didn't think it was possible to do anything but Initialised -> DepartureSubmitted.  A Departure Declaration is the only message that can create a Departure movement.  @RobertLaverick am I thinking wrong?

<img width="531" alt="Screenshot 2021-04-13 at 15 12 18" src="https://user-images.githubusercontent.com/1506327/114566903-a5f9a580-9c6a-11eb-8cf6-2b05c1f601ad.png">

Reply:
There are a few transitions that are accepted by the code currently that aren't on the tube maps so I was wondering whether we wanted to make this validation more strict

Reply:
@crystalmark these are the transitions that are permitted by the current code: https://bit.ly/3wZ72K5

I used the [Quiver](https://verizon.github.io/quiver/) library

via
```scala
import models._
import quiver._

val nodes: Seq[UNode[DepartureStatus]] = 
  DepartureStatus.values.map(LNode(_, ()))

val edges: Seq[UEdge[DepartureStatus]] = StatusTransition.validTransitions
  .filterNot { case (from, to) => from == to }
  .map { case (from, to) => LEdge(from, to, ()) }
  .toSeq

val transitionGraph = mkGraph(nodes, edges)

val graphviz: String = quiver.viz.graphviz(transitionGraph, "Departure Status Transitions")
```